diff --git a/source/Irrlicht/Makefile b/source/Irrlicht/Makefile
index 67a3899..3699f10 100644
--- a/source/Irrlicht/Makefile
+++ b/source/Irrlicht/Makefile
@@ -1,6 +1,9 @@
 VERSION_MAJOR = 1
 VERSION_MINOR = 8
 VERSION_RELEASE = 4
+
+all: sharedlib
+
 # Irrlicht Engine 1.8.4
 # Makefile for Linux
 #
@@ -86,7 +89,7 @@ endif
 staticlib sharedlib install: SYSTEM = Linux
 STATIC_LIB = libIrrlicht.a
 LIB_PATH = ../../lib/$(SYSTEM)
-INSTALL_DIR = /usr/local/lib
+INSTALL_DIR = $(DESTDIR)/usr/lib64
 sharedlib install: SHARED_LIB = libIrrlicht.so
 sharedlib: LDFLAGS += -L/usr/X11R6/lib$(LIBSELECT) -lGL -lXxf86vm
 staticlib sharedlib: CXXINCS += -I/usr/X11R6/include
